ballerina/http_request_context.bal

```diff
@@ -15,10 +15,13 @@
 // under the License.
 
 import ballerina/jballerina.java;
-import ballerina/lang.value;
+
+// This is same as the `value:Cloneable`, except that it does not include `error` type.
+# Represents a non-error type that can be cloned.
+public type Cloneable (any & readonly)|xml|Cloneable[]|map<Cloneable>|table<map<Cloneable>>;
```

Comment:
This will also disallow `error[]`, `map<error>`, etc, although they are disjoint with the error type. Hope that's OK.

Reply:
If we want to allow them, we can probably use 

```suggestion
public type Cloneable (any & readonly)|xml|value:Cloneable[]|map<value:Cloneable>|table<map<value:Cloneable>>;
```

Reply:
Most of the time `RequestContext` is used to store the `any data` type. But we kept it to support all the general types. IMO we don't need to support structure types of `error`. (But anyway I will update the spec with this information)
@chamil321 @shafreenAnfar WDYT?

Reply:
Yeah, I also think that passing around error is not useful, therefore, lets revisit this if there is some requirement .